platforms/common/CMakeLists.txt

```diff
@@ -51,9 +51,10 @@ add_library(px4_platform STATIC
 	px4_cli.cpp
 	shutdown.cpp
 	spi.cpp
+	Serial.cpp
 	${SRCS}
 )
-target_link_libraries(px4_platform prebuild_targets px4_work_queue)
+target_link_libraries(px4_platform prebuild_targets px4_work_queue px4_layer)
```

Comment:
I checked and this is causing the redefinition errors.

Reply:
The whole point is that there is a common part of the driver (Serial.cpp) and a platform specific part of the driver (SerialImpl.cpp). Each platform has it's own platform specific version of SerialImpl.cpp.

Reply:
The redefinition errors are only occurring with the protected build, not the default build. So there is something about the linking for that type of build that requires a different approach.

Reply:
I was able to resolve the issue in the latest commit.